packages/experimental/Switch/src/useSwitch.ts

```diff
@@ -0,0 +1,72 @@
+import * as React from 'react';
+import { useAsPressable, useKeyProps, useOnPressWithFocus, useViewCommandFocus } from '@fluentui-react-native/interactive-hooks';
+import { SwitchProps, SwitchInfo } from './Switch.types';
+import { LayoutAnimation } from 'react-native';
+import { InteractionEvent } from '@fluentui-react-native/interactive-hooks';
+
+const defaultAccessibilityActions = [{ name: 'Toggle' }];
+
+export const useSwitch = (props: SwitchProps): SwitchInfo => {
+  const defaultComponentRef = React.useRef(null);
+  const {
+    onChange,
+    checked,
+    defaultChecked,
+    label,
+    labelPosition,
+    componentRef = defaultComponentRef,
+    disabled,
+    accessibilityRole,
+    ...rest
+  } = props;
+
+  const isDisabled = !!disabled;
+  const initialCheckedState = !!(checked ?? defaultChecked);
+  const [checkedState, setCheckedState] = React.useState(initialCheckedState);
+  const [prevCheckedState, setPrevCheckedState] = React.useState(initialCheckedState);
+  const focusRef = isDisabled ? null : componentRef;
+
+  if (defaultChecked !== undefined && checked !== undefined) {
+    console.warn('The props defaultChecked and checked are mutually exclusive. Use only one of the props, do not use both.');
+  }
+
+  const toggleCallback = React.useCallback(
+    (e: InteractionEvent) => {
+      const newCheckedState = checked !== undefined ? checked : !checkedState;
+      onChange && onChange(newCheckedState, e);
+      setCheckedState(newCheckedState);
+    },
+    [onChange, setCheckedState, checkedState, checked],
+  );
```

Comment:
Instead of using your own callback for `toggleCallback`, could you use `useAsToggleWithEvent` under interactive-hooks? This would help maintain the "real" checked state like you're tracking with `currentCheckedState` but without the need for the extra variable.

Reply:
I've made it use `useAsToggleWithEvent `, it cleaned up things well thanks!

Reply:
Glad it worked!